packages/vike-react/renderer/getLang.ts

```diff
@@ -0,0 +1,42 @@
+export { getLang }
+
+import type { PageContext } from 'vike/types'
+import { isCallable } from './utils/isCallable.js'
+
+/**
+ * Get the page's lang if defined, either from the config, the additional data fetched by
+ * the page's data() and onBeforeRender() hooks or from other hooks.
+ */
+function getLang(pageContext: PageContext): null | string {
+  // from data() hook
+  if (pageContext.data?.lang !== undefined) {
+    return pageContext.data.lang
+  }
+
+  // TODO/next-major-release: remove support for setting lang over onBeforeRender()
+  // from onBeforeRender() hook & other hooks, e.g. onBeforeRoute() hook
+  if (pageContext.lang !== undefined) {
+    return pageContext.lang
+  }
```

Comment:
We don't want to deprecate this as this is precisely how it will be set by the hooks covered in https://vike.dev/i18n . So I think we shouldn't mention here `onBeforeRender()` but rather `onBeforeRoute()`

Reply:
Oh I see you mention `onBeforeRoute()` already, but yeah I wouldn't mention `onBeforeRender()` here and remove the fist comment line (`TODO/...`)

Reply:
Done